media_driver/linux/common/os/hwinfo_linux.c

```diff
@@ -133,6 +133,21 @@ MOS_STATUS HWInfo_GetGfxInfo(int32_t           fd,
         return MOS_STATUS_PLATFORM_NOT_SUPPORTED;
     }
 
+    if (gtSystemInfo->VDBoxInfo.NumberOfVDBoxEnabled == 0)
+    {
+        unsigned int nengine = MAX_ENGINE_INSTANCE_NUM;
+        struct i915_engine_class_instance uengines[MAX_ENGINE_INSTANCE_NUM];
+        if (mos_query_engines(fd,I915_ENGINE_CLASS_VIDEO,0,&nengine,uengines) == 0)
+        {
+            gtSystemInfo->VDBoxInfo.NumberOfVDBoxEnabled = nengine;
+        }
+        else
+        {
+            //enable 1 vdbox by default
+            gtSystemInfo->VDBoxInfo.NumberOfVDBoxEnabled = 1;
```

Comment:
Why this? It sounds more logical to trust the query - but if query failed then other places in the code seem to trigger failure rather faking data? It could be some code base quirk I do not understand of course...

Reply:
At the beginning, wanted to keep the code workable even when the GT query patch is not available.